docs/usage/shoot_pod_autoscaling_best_practices.md

```diff
@@ -0,0 +1,329 @@
+
+VPA is very useful for optimizing the resource requests of pods that have dynamic resource needs over time. It does so by mutating pod requests (unfortunately, [not in-place](https://github.com/kubernetes/design-proposals-archive/blob/main/autoscaling/vertical-pod-autoscaler.md#in-place-updates)). Therefore, in order to apply new recommendations, pods that are "out of bounds" (i.e. below a configured/computed lower or above a configured/computed upper recommendation percentile) will be evicted proactively, but also pods that are "within bounds" may be evicted after a grace period. The corresponding higher-level replication controller will then recreate a new pod that VPA will then mutate to set the currently recommended requests (and proportional limits, if configured).
+
+> [!NOTE]
+> VPA continuously monitors all targeted pods and calculates recommendations based on their usage (one recommendation for the entire target). This calculation is influenced by configurable percentiles, with a greater emphasis on recent usage data and a gradual decrease (=decay) in the relevance of older data. However, this means, that VPA doesn't take into account individual needs of single pods - eventually, all pods will receive the same recommendation, which may lead to considerable resource waste. Ideally, VPA would update pods [in-place](https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/1287-in-place-update-pod-resources/README.md) depending on their individual needs, but that's (individual recommendations) not in its design, even if [in-place updates](https://github.com/kubernetes/design-proposals-archive/blob/main/autoscaling/vertical-pod-autoscaler.md#in-place-updates) get implemented, which may be years away for VPA based on current activity on the component.
+
+## Selecting the Appropriate Autoscaler
+
+Before deciding on an autoscaling strategy, it's important to understand the characteristics of your application:
+- **Statelessness:** Is your application stateless or stateful? Stateless applications are typically better candidates for HPA as they can be easily scaled out by adding more replicas without worrying about maintaining state.
+- **Traffic Patterns:** Does your application experience variable traffic? If so, HPA can help manage these fluctuations by adjusting the number of replicas to handle the load.
+- **Resource Usage:** Does your application's resource usage change over time? VPA can adjust the CPU and memory allocations dynamically, which is beneficial for applications with non-uniform resource requirements.
+- **Scalability:** Can your application handle increased load by scaling vertically (more resources per pod) or does it require horizontal scaling (more pod instances)?
```

Comment:
An additional point to consider is: how sensible are clients to disruptions? Even if your workload is stateless (like kube-apiserver), we still see complaints from people if single connections get interrupted and clients would need to retry.
Not sure if we want to argue in one direction or the other:
Either: If you want to use VPA, make sure your clients retry on short disruptions
Or: Don't consider using VPA if your clients are very sensible to disruptions

Reply:
Ah, that's an excellent point I completely missed to add. I will do that! Thanks @voelzmo!

Reply:
OK, I added a new section on interruptibility to the top of the list. Thank you @voelzmo!